src/atom-environment.js

```diff
@@ -182,7 +180,18 @@ class AtomEnvironment {
       default: path.join(fs.getHomeDirectory(), 'github'),
       description: 'The directory where projects are assumed to be located. Packages created using the Package Generator will be stored here by default.'
     }
-    this.config.initialize({configDirPath: this.configDirPath, resourcePath, projectHomeSchema: ConfigSchema.projectHome})
+    if (this.enablePersistence) {
+      this.configStorage = new ConfigStorage({
+        config: this.config,
+        configDirPath: this.configDirPath,
+        resourcePath,
+        notificationManager: this.notifications
+      })
+    }
+    this.config.initialize({
+      configFilePath: ConfigStorage.createConfigFilePath(this.configDirPath),
+      projectHomeSchema: ConfigSchema.projectHome
+    })
```

Comment:
@damieng I merged `master` into this branch so you wouldn't have to deal with the conflicts I created in converting `atom-environment.coffee` to JS. Luckily your changes to `atom-environment.coffee` were pretty small so it was easy to bring them over to `atom-environment.js`. You might want to give these a 👀  though, just in case I missed something.